tests/queries/0_stateless/02946_materialize_column_must_not_override_past_values.sql

```diff
@@ -0,0 +1,53 @@
+SET mutations_sync = 2;
+
+SELECT '-- Compact parts';
+
+CREATE TABLE tab (id Int64, dflt Int64 DEFAULT 54321) ENGINE MergeTree ORDER BY id;
+INSERT INTO tab (id, dflt) VALUES (1, 2);
```

Comment:
As far as I understand, the logic is based on the fact that a part as a whole may or may not contain default values for a column. What is not possible (and there seems no syntax for that) is to have a part which contains default and non-default values. Is my understanding right?

Reply:
>  What is not possible (and there seems no syntax for that) is to have a part which contains default and non-default values. Is my understanding right?

Yes, I cannot think of any case that a column in a part may contains both default and non-default values. Only potential cases I could think about is DELETE/UPDATE, but mutation will rewrite the whole part and thus, will materialise the column if it's absent.